pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadUtils.java

```diff
@@ -119,12 +123,61 @@ public InputStream createInputStream() throws IOException {
   }
 
   public static int sendSegmentFile(final String host, final String port, final String fileName,
+      File file, final long lengthInBytes) {
+    return sendSegmentFile(host, port, fileName, file, lengthInBytes, MAX_RETRIES, SLEEP_BETWEEN_RETRIES_IN_SECONDS);
+  }
+
+  public static int sendSegmentFile(final String host, final String port, final String fileName,
+      File file, final long lengthInBytes, int maxRetries, int sleepTimeSec) {
+    for (int numRetries = 0; ; numRetries++) {
+      try {
+        InputStream inputStream = new FileInputStream(file);
```

Comment:
Put this inside try (InputStream inputStream = new FileInputStream(file)) to prevent leak.